### ISSUE TITLE
Wrong line number after multiline close group

### ISSUE THREAD

**Issue by RockinRoel** [doxygen/doxygen#7393]
When closing a group on more than one line, like:

```cpp
/*! @}
 */
```

The line count gets shifted after that (one line less than it should be).

Here's an example file:

```cpp
/*! \class DocumentThis
 *  \brief Class desc
 */
class DocumentThis {
public:

    /*! \brief Ctor
     */
    DocumentThis();

    /*! @name Group one
     * @{
     */

    /*! \brief Method one
     */
    void methodOne();

    /*! @}
     */

    /*! @name Group two
     * @{
     */

    /*! \brief Method two
     */
    void methodTwo();

    /*! @}
     */
};
```

If you use a Doxyfile with `GENERATE_XML` set to `YES`, then you will get a line number for `methodTwo` of 27, instead of the expected 28.

Adding a `lineCount();` call after the `Doxygen::docGroup.close(...)` in `src/scanner.l` seems to fix this. I'm not familiar enough with doxygen or flex to know for sure if that's right, though.

**Comment by albert-github** [comment]
Line counting is a very difficult and tricky matter.

- which version are you using
- which version of the code are you referring to for the fix? Do you refer to the lines in scanner.l:
  ```
    <FindMembers,FindFields,ReadInitializer>"//"([!/]){B}*{CMD}"}".*|"/*"([!*]){B}*{CMD}"}"[^*]*"*/"        {
                                          bool insideEnum = YY_START==FindFields || (YY_START==ReadInitializer && lastInitializerContext==FindFields); // see bug746226
                                          Doxygen::docGroup.close(current.get(),yyFileName,yyLineNr,insideEnum);
                                        }
  ```

I did a small test and it looks OK to me, also in the doxygen tests (`make tests`) I don't see problems, though this problem is not tested in here (`019_defgroup.c`).

Problem should not only be tested against `@name` but also against `@defgroup` etc., did this and looks OK as well.

Do you want to create a proposed pull request for it or should I do this.

**Comment by RockinRoel** [comment]
I was referring to the `master` branch, though I had noticed it first in the doxygen-1.8.13-10 package on Ubuntu 18.04. That's indeed the code I'm referring to. If you think my fix is appropriate, I can make a PR for it.

**Comment by albert-github** [comment]
Yes please make a proposed pull request.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
